### PR TITLE
Milestone 7:- Add Unit Tests

### PR DESCRIPTION
**Pull request by hmunish**
- Add tests for person, student, teacher, book & classroom classes.

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    it 'should create instance with correct title' do
+      expect(book.title).to eq 'Harry Potter'
+    end
+
+    it 'should create instance with correct author name' do
+      expect(book.author).to eq 'J.K.Rowlings'
+    end
+  end
+end
```
Comment (KanzaTahreem) [inline spec/book_spec.rb]: - [x] Great implementation so far 👏👏 However, please make sure to test the methods present in the `Book` class as mentioned in the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md)


image

File: spec/decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  end
+end
+describe TrimmerDecorator do
+  context 'When testing TrimmerDecorator class' do
+    trimmer_decorator = TrimmerDecorator.new('Jimmy')
+    it 'should create instance of trimmer_decorator' do
+      expect(trimmer_decorator).to be_an_instance_of TrimmerDecorator
+    end
+  end
+end
```
Comment (KanzaTahreem) [inline spec/decorator_spec.rb]: - [x] Here again, please make sure to test the `correct_name` method present in all of these classes as mentioned in the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md)
- [x] Also, kindly make sure to test `Nameable`  class that you created while implementing the Decorator design pattern as per requirements 💪 for `Nameable`, you could try 👇

```ruby
describe Nameable do
  before :each do
    @nameable = Nameable.new
  end

  context 'When testing a Nameable class' do
    it 'should be an instance of Nameable' do
      expect(@nameable).to be_an_instance_of Nameable
    end
  end

  context 'When using correct_name method' do
    it 'should raise an error' do
      expect { @nameable.correct_name }.to raise_error('NotImplementedError')
    end
  end
end
```


image

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    it 'should return name of person instance' do
+      expect(person.name).to eq 'Jimmy'
+    end
+
+    it 'should return true' do
+      expect(person.can_use_services?).to be true
+    end
+  end
+end
```
Comment (KanzaTahreem) [inline spec/person_spec.rb]: - [x] Here as well, please make sure to test the remaining methods present in the `Person` class as mentioned in the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md)


image

File: spec/student_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    it 'should return name of Student instance' do
+      expect(student.name).to eq 'Jimmy'
+    end
+
+    it 'should return true' do
+      expect(student.play_hooky).to eq '¯(ツ)/¯'
+    end
+  end
+end
```
Comment (KanzaTahreem) [inline spec/student_spec.rb]: - [x] Here again, please make sure to test the `classroom` present in the `Student` class as mentioned in the [project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md)


image

**Review by KanzaTahreem — CHANGES_REQUESTED** [review]
# Changes Required 🔁

Hi @hmunish & @abdozayan12 

Good job so far 👏! I really appreciate the effort you put in!  **_However, there are few this that still need to be fixed. But you are almost there!_** 💪 🏁

so close


## Highlights 🎯

There are some excellent points that you have made perfectly. Keep up the great work! 👏
- ✅ Tests are placed in `/spec` dir
- ✅ All implemented tests are passing

## Required Changes 🛑

_Please check the comments under the review._

## Optional suggestions ❕

N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tugsem — APPROVED** [review]
## STATUS: APPROVED 👍🎯  
Hello @hmunish 👋 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⭐ 
✔️ Required tests added
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**